legal-api/src/legal_api/resources/v2/business/business_filings/business_filings.py

```diff
@@ -539,13 +540,30 @@ def check_authorization(identifier, filing_json: dict,
         # for incorporationApplication and registration, get legalType from nameRequest
         else:
             legal_type = filing_json['filing'][filing_type]['nameRequest'].get('legalType')
-
-        if not authorized(identifier, jwt, action=['edit']):
-            return jsonify({'message': f'You are not authorized to submit filings for {identifier}.'}), \
+        if filing_sub_type:
+            filing_type = f'{filing_type}.{filing_sub_type}'
+        if (
+            flags.is_on('enable-permissions-for-action')
+            and not PermissionService.has_permissions_for_action(filing_type)
+        ):
+            return (
+                jsonify({
+                    'message': (
+                        f'Permission Denied - You do not have permissions to submit this type of filing for: '
+                        f'{identifier}.'
+                    )
+                }),
                 HTTPStatus.FORBIDDEN
-
+            )
+        if not authorized(identifier, jwt, action=['edit']):
+            return jsonify({
+                'message': f'Not Authorized - You are not permitted to submit filings for {identifier}.'
```

Comment:
Change the error message to "... You are not authorized ..." (to match the condition you're checking).